src/main/java/org/sonarlint/intellij/plsql/PLSQLLanguageActivator.java

```diff
@@ -26,8 +26,11 @@
 public class PLSQLLanguageActivator implements LanguageActivator {
 
   @Override
-  public void amendLanguages(Set<Language> enabledLanguages) {
-    enabledLanguages.add(Language.PLSQL);
+  public void amendLanguages(Set<Language> enabledLanguages, boolean isConnected) {
+    boolean isAlreadyEnabled = enabledLanguages.contains(Language.PLSQL);
```

Comment:
I don't see a situation where it would already be enabled. Do you really need to check that?
Also `enabledLanguages` is a Set, so there won't be any duplication if you add it twice.
I suggest removing the `!isAlreadyEnabled` part in the if below